cli/src/bin/phylum.rs

```diff
@@ -180,45 +181,68 @@ async fn handle_commands() -> CommandResult {
         .await;
     }
 
-    let mut api = PhylumApi::new(
-        &mut config.auth_info,
-        &config.connection.uri,
-        timeout,
-        ignore_certs,
-    )
-    .await
-    .context("Error creating client")?;
-
-    // PhylumApi may have had to log in, updating the auth info so we should save the config
-    save_config(&config_path, &config).with_context(|| {
-        format!(
-            "Failed to save configuration to '{}'",
-            config_path.to_string_lossy()
-        )
-    })?;
+    // This is `Future<Output = PhylumApi>`. Commands that require access to the API
+    // will await on this, so that the API is not instantiated ahead of time for subcommands
+    // that don't require it.
```

Comment:
I think the comment is mentioning the type in an effort to point out that we are deliberately using the future here rather than immediately awaiting. Something like this might be more clear.

```suggestion
    // Creating the PhylumAPI object requires that the user be authenticated. But not
    // every command needs authentication. So here we intentionally delay awaiting on
    // the future in order to also delay the authentication requirement.
```

Reply:
I agree on both points, but on one hand we can't make the type explicit AFAIK (as it's an opaque `impl Future<...>`) and on the other there may be a real need to address this in some way, as I interpreted from @kylewillmon's observation that the `async` block felt weird in the sense that it's an unawaited `Future`. I feel like having the comment is a decent compromise, but if there was another way of expressing the type inline that's tolerable by Rust I'd go that way instead.

Reply:
I think his point is only that mentioning the type is unnecessary. The comment is definitely helpful. But it's just as helpful without the type:

```suggestion
    // Commands that require access to the API will await on this, so that the API is not
    // instantiated ahead of time for subcommands that don't require it.
```

Or if you prefer, just replace the mention of the type with something like this:

```suggestion
    // Get the future, but don't await. Commands that require access to the API will await on this,
    // so that the API is not instantiated ahead of time for subcommands that don't require it.
```